File: cci_1-1.cc
//================================================
// Cracking the coding interview - Exercise 1.1
// March 3rd, 2017
//================================================

#include <iostream>
#include <vector>
#include <stdlib.h>

int main()
{// This program tests a provided string to determine if all chars are unique
    std::string testString;
    std::vector<int> charFrequency(26, 0);

    std::cout << "Please enter the string to test for duplicate chars: ";
    std::cin >> testString;

    for (auto c: testString) {
        charFrequency[c - 'a']++;
        if (charFrequency[c - 'a'] > 1) {
            std::cout << "\nString does not contain all unique characters\n";
            return 0;
        }
    }

    std::cout << "\nString contains all unique characters - success!\n";
    return 0;
}

File: cci_1-1.py
"""
Cracking the coding interview - Exercise 1.1
March 3rd, 2017
"""
import sys
from collections import Counter


def count_unique(testString):
    """
    This function determines if every character in a provided string
    is unique. Exits if yes or returns none if no
    """
    charFrequency = Counter(testString)

    for i in charFrequency.values():
        if i > 1:
            sys.exit("\nString does not contain all unique characters\n")


testString = input("\nPlease enter the string to test for duplicate chars: ")

count_unique(testString)

print("\nString contains all unique characters - success!\n")
